### ISSUE TITLE
blobfuse does not function with or respect chown

### ISSUE THREAD

**Issue by austingsmith** [Azure/azure-storage-fuse#564]
### Which version of the blobfuse was used?
latest

### Which OS (please include version) are you using?
ubuntu 18.04

### What problem was encountered?
blobfuse overwrites the directory ownership contrary to the documentation relating to ownership, essentially making the app unusable due to the web service not being able to read files owned by another user (root or the non-priv acct, web service being nologin)

### Have you found a mitigation/solution?
YEAP - use something else and ditch this solution.

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?
nope, the mount is successful, but it changes the ownership of the directory when mounted and while chown completes, the uid/gid ownership changes NOT.

### If relevant, please share your mount command.

root@vmss-gtnw000001:/var/www# chown www-data:www-data html
root@vmss-gtnw000001:/var/www# ls -al
total 8
drwxr-xr-x  3 root root 4096 Feb  5 19:22 .
drwxr-xr-x 14 root root 4096 Feb  5 19:22 ..
drwxrwx---  2 root root 4096 Feb  5 21:05 html
root@vmss-gtnw000001:/var/www# id www-data
uid=33(www-data) gid=33(www-data) groups=33(www-data)
root@vmss-gtnw000001:/var/www# umount html
root@vmss-gtnw000001:/var/www# ls -al
total 12
drwxr-xr-x  3 root root 4096 Feb  5 19:22 .
drwxr-xr-x 14 root root 4096 Feb  5 19:22 ..
drwxr-xr-x  2 root root 4096 Feb  5 20:51 html
root@vmss-gtnw000001:/var/www# chown www-data:www-data html
root@vmss-gtnw000001:/var/www# ls -al
total 12
drwxr-xr-x  3 root     root     4096 Feb  5 19:22 .
drwxr-xr-x 14 root     root     4096 Feb  5 19:22 ..
drwxr-xr-x  2 www-data www-data 4096 Feb  5 20:51 html
root@vmss-gtnw000001:/var/www# blobfuse /var/www/html --tmp-path=/mnt/blobtmp --config-file=/etc/fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120
root@vmss-gtnw000001:/var/www# ls -al
total 8
drwxr-xr-x  3 root root 4096 Feb  5 19:22 .
drwxr-xr-x 14 root root 4096 Feb  5 19:22 ..
drwxrwx---  2 root root 4096 Feb  5 21:34 html
root@vmss-gtnw000001:/var/www#

**Comment by vibhansa-msft** [comment]
Blobfuse does not support 'Chown'. '

**Comment by NaraVen** [comment]
@austingsmith , could you try mounting with the -o allow_other option? your webservice app should be able to read the mount files if you do that. The only risk in allow_other is that you are giving permissions to all users to read.

**Comment by andyzhangx** [comment]
`-o allow_other` option would set permission to 0777, while it does not mean it supports `chown` which changes the group id and user id of file or directory, below example shows `chown` & `chmod` is not supported in blobfuse mount:
```
# k exec -it statefulset-blob-0 sh
# df -h
Filesystem      Size  Used Avail Use% Mounted on
blobfuse         14G   46M   13G   1% /mnt/blob
# cd /mnt/blob
# ls -lt
total 5952
-rwxrwxrwx 1 root root 6090986 May 28 09:07 outfile
# chown 100:100 outfile
# ls -lt
total 5952
-rwxrwxrwx 1 root root 6091363 May 28 09:07 outfile
# chmod 0653 outfile
# ls -lt
total 0
-rwxrwxrwx 1 root root 6091769 May 28 09:08 outfile
```

**Comment by NaraVen** [comment]
yes, to be clear chown is not supported and Only adls gen2 supports chmod. In my comment above for regular blob I suggested -o allow_other so that users and groups that don't normally have permission get these permissions granted in the mount dir.

**Comment by vibhansa-msft** [comment]
As chown is not supported by azure storage itself, we can not support it from blobfuse end. Closing this thread.
